### ISSUE TITLE
cumsum not same behaviour

### ISSUE THREAD

**Issue by diorcety** [serge-sans-paille/pythran#709]
Here the code for reproduce
```
#pythran export f(float[])
#pythran export f(float[][])
def f(x):
    return np.cumsum(x, axis=0)
```

```
import diff
import numpy as np
from math import factorial

ROI_size = 3 
N = ROI_size**2

sample_spacing = 0.0000001
p = np.arange(0.,1.,sample_spacing)

sample_size = len(p)

proba = np.zeros( (N+1, sample_size) )
for i in xrange(N+1):
   x = 2 
   binomial_coef = 1. * factorial(N) / factorial(i) / factorial(N-i)
   proba[i, x] = binomial_coef * p[x]**i * (1-p[x])**(N-i)

print(["%.32f" % x for x in proba[:,2][0:10]])
cumulated = diff.f(proba)
print(["%.32f" % x for x in cumulated[:,2][0:10]])
```

Python output

> ['0.99999820000143990750274269885267', '0.00000179999712000201578593274084', '0.00000000000143999798400120935055', '0.00000000000000000067199919360040', '0.00000000000000000000000020159980', '0.00000000000000000000000000000004', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000']
> ['0.99999820000143990750274269885267', '0.99999999999855992971475870945142', '0.99999999999999988897769753748435', '0.99999999999999988897769753748435', '0.99999999999999988897769753748435', '0.99999999999999988897769753748435', '0.99999999999999988897769753748435', '0.99999999999999988897769753748435', '0.99999999999999988897769753748435', '0.99999999999999988897769753748435']

Pythran output

> ['0.99999820000143990750274269885267', '0.00000179999712000201578593274084', '0.00000000000143999798400120935055', '0.00000000000000000067199919360040', '0.00000000000000000000000020159980', '0.00000000000000000000000000000004', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000']
> ['0.99999820000143990750274269885267', '0.99999820000143990750274269885267', '0.99999820000143990750274269885267', '0.99999820000143990750274269885267', '0.99999820000143990750274269885267', '0.99999820000143990750274269885267', '0.99999820000143990750274269885267', '0.99999820000143990750274269885267', '0.99999820000143990750274269885267', '0.99999820000143990750274269885267']

**Comment by diorcety** [comment]
```
import diff
import numpy as np

proba = np.zeros( (10, 10000) )
proba[:,4] = np.arange(0,10,1)

print(["%.32f" % x for x in proba[:,2][0:10]])
cumulated = diff.f(proba)
print(["%.32f" % x for x in cumulated[:,2][0:10]])

proba = np.zeros( (10, 1000) )
proba[:,4] = np.arange(0,10,1)

print(["%.32f" % x for x in proba[:,2][0:10]])
cumulated = diff.f(proba)
print(["%.32f" % x for x in cumulated[:,2][0:10]])
```

Output:

> ['0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000']
> ['0.00000000000000000000000000000000', '1.00000000000000000000000000000000', '3.00000000000000000000000000000000', '6.00000000000000000000000000000000', '10.00000000000000000000000000000000', '15.00000000000000000000000000000000', '21.00000000000000000000000000000000', '28.00000000000000000000000000000000', '36.00000000000000000000000000000000', '45.00000000000000000000000000000000']
> ['0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000']
> ['0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000', '0.00000000000000000000000000000000']

**Comment by diorcety** [comment]
Following the size of the array, the result is not the same ... very strange

**Comment by diorcety** [comment]
with proba = np.zeros( (10, 100000) ) i have a segmentation fault with pythran, and no issue at all with python

**Comment by diorcety** [comment]
https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.ufunc.accumulate.html

For example, add.accumulate() is equivalent to np.cumsum().
replace np.cumsum by np.add.accumulate() in this example and all is working great

**Comment by diorcety** [comment]
Replace iadd by add 
https://github.com/serge-sans-paille/pythran/blob/master/pythran/pythonic/numpy/cumsum.hpp#L21

**Comment by serge-sans-paille** [comment]
Fixed in #712
